src/cargo/ops/cargo_rustc/context/mod.rs

```diff
@@ -208,11 +208,17 @@ impl<'a, 'cfg> Context<'a, 'cfg> {
             jobserver,
             build_script_overridden: HashSet::new(),
 
-            unit_dependencies: None,
+            unit_dependencies: HashMap::new(),
             // TODO: Pre-Calculate these with a topo-sort, rather than lazy-calculating
             target_filenames: HashMap::new(),
             target_metadatas: HashMap::new(),
-        })
+        };
+
+        cx.probe_target_info()?;
+        let deps = build_unit_dependencies(units, &cx)?;
+        cx.unit_dependencies = deps;
```

Comment:
This is a funny thing to do in constructor... Ideally, we do something like `unit_dependencies: build_unit_dependencies(units, some other arguments)`, but this is not trivial to do at this stage  :)